slack.ts

```diff
@@ -0,0 +1,54 @@
+/** Error20:
+    - line 21: don't know how to declare call signature in tsc*/
+
+import * as querystring from "querystring";
+import fetch from "node-fetch";
+
+const tkn = process.env.SLACK_TOKEN;
+const domain = process.env.DOMAIN;
+
+async function sendMessage(message : string, config : string) : Promise<void> {
+    const msg = {
+        token: tkn,
+        channel: config,
+        as_user: false,
+        username: "HackGT",
+        text: message
+    };
+
+    const qs = querystring.stringify(msg);
+
+    const res = await
+        fetch(`https://${domain}.slack.com/api/chat.postMessage?${qs}`);
+    const json = await res.json();
+    console.log(json);
+}
+
+async function check(config : any): Promise<string> {
+    const res = await
+        fetch(`https://${domain}.slack.com/api/channels.list`);
+    const channels = await res.json();
+
+    const res2 = await
+        fetch(`https://${domain}.slack.com/api/groups.list`);
+    const groups = await res2.json();
+
+    var resultsChannels : string[] = [];
+    var resultsGroups : string[] = [];
+    var searchField = 'name';
+    var searchValue = config;
+
+    for (var i = 0; i < channels.length; i++) {
```

Comment:
Easier way is
```typescript
if (resultsChannels.map(c => c.name).includes(searchValue)) {
  return searchValue;
}
```

Reply:
What does `c => c.name` indicate?

Reply:
It's a new syntax that lets you declare an anonymous function, the left side is the arguments to the function (just `c`), the right is the function body with an implicit `return` statement. You can think of it as this:
```typescript
// can be written shorter with `c => c.name`
function(c) {
  return c.name;
}
```

Reply:
Oh and `map` might be unfamiliar too. It takes every element of an iterator and applies a function on it, returning a list of elements with a function applied to them, so the following takes the square of every element:
```typescript
const squares = [1, 2, 3, 4].map(n => n * n); // equals [1, 4, 9, 16]
```